builtin/providers/aws/resource_aws_route53_record.go

```diff
@@ -68,13 +66,10 @@ func resourceAwsRoute53Record() *schema.Resource {
 				ConflictsWith: []string{"alias"},
 			},
 
-			// Weight uses a special sentinel value to indicate it's presense.
-			// Because 0 is a valid value for Weight, we default to -1 so that any
-			// inclusion of a weight (zero or not) will be a usable value
 			"weight": &schema.Schema{
 				Type:     schema.TypeInt,
 				Optional: true,
-				Default:  -1,
```

Comment:
We can address this in a migration, e.g:

- https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_security_group_rule_migrate.go

We'll need to do that for all `-1` entries.